File: auth-service/src/main/java/lib/quick/authservice/global/security/details/MemberDetails.java
package lib.quick.authservice.global.security.details;

import lib.quick.authservice.domain.member.entity.Member;
import lombok.AllArgsConstructor;
import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.authority.SimpleGrantedAuthority;
import org.springframework.security.core.userdetails.UserDetails;
import java.util.Collection;
import java.util.List;

@AllArgsConstructor
public class MemberDetails implements UserDetails {
    private Member member;

    @Override
    public Collection<? extends GrantedAuthority> getAuthorities() {
        return List.of(new SimpleGrantedAuthority(member.getRole().getName()));
    }

    @Override
    @Deprecated
    public String getPassword() {
        return null;
    }

    @Override
    public String getUsername() {
        return member.getEmail();
    }
}

File: auth-service-spring/src/main/kotlin/lib/quick/authservice/global/security/details/MemberDetails.kt
package lib.quick.authservice.global.security.details;

import lib.quick.authservice.domain.user.entity.User
import org.springframework.security.core.GrantedAuthority
import org.springframework.security.core.authority.SimpleGrantedAuthority;
import org.springframework.security.core.userdetails.UserDetails

class MemberDetails(
    private val user: User
): UserDetails {

    override fun getAuthorities(): MutableCollection<out GrantedAuthority> {
        return mutableListOf(SimpleGrantedAuthority(user.role.name));
    }

    override fun getPassword(): String? {
        return null;
    }

    override fun getUsername(): String {
        return user.email;
    }
}
